openshift/release/mirror-upstream-branches.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Usage: openshift/release/mirror-upstream-branches.sh
+# This should be run from the basedir of the repo with no arguments
+
+
+set -ex
+readonly TMPDIR=$(mktemp -d knativeEventingBranchingCheckXXXX -p /tmp/)
+
+git fetch upstream --tags
+git fetch openshift --tags
+
+# We need to seed this with a few releases that, otherwise, would make
+# the processing regex less clear with more anomalies
+cat >> "$TMPDIR"/midstream_branches <<EOF
+0.2
+0.3
+EOF
+
+git branch --list -a "upstream/release-1.*" | cut -f3 -d'/' | cut -f2 -d'-' > "$TMPDIR"/upstream_branches
+git branch --list -a "openshift/release-v1.*" | cut -f3 -d'/' | cut -f2 -d'v' | cut -f1,2 -d'.' >> "$TMPDIR"/midstream_branches
+
+sort -o "$TMPDIR"/midstream_branches "$TMPDIR"/midstream_branches
+sort -o "$TMPDIR"/upstream_branches "$TMPDIR"/upstream_branches
+comm -32 "$TMPDIR"/upstream_branches "$TMPDIR"/midstream_branches > "$TMPDIR"/new_branches
+
+UPSTREAM_BRANCH=$(cat "$TMPDIR"/new_branches)
```

Comment:
sure that there will be only 1 new upstream branch ?  Looks like a hard assumption.

Maybe exit with an error or use a `head -1` instead of `cat`

Reply:
That's a valid point. There will a few new_branches at least for the first ever execution.

Reply:
WDYT? https://github.com/openshift/knative-client/pull/1134/commits/da9e8c4ed2c0eb00a771740e9928f575c109a3cc